### ISSUE TITLE
Superinterface of a TypeSpec

### ISSUE THREAD

**Issue by kenzierocks** [square/javapoet#411]
Currently we can reference other `TypeSpec`s by name using `$T`, but it doesn't seem that the same applies for declaring a `TypeSpec` as a superinterface. Is this something that could be added with the current model, or do I just do `ClassName.get(spec.name)`?

**Comment by swankjesse** [comment]
The catch is that a TypeSpec doesn’t know if it’s a top-level type, and it doesn’t know its own package name.

**Comment by kenzierocks** [comment]
Yes, that's what I was thinking. Do you think this is something you'd like to support?

**Comment by swankjesse** [comment]
It’s awkward to give a nested class TypeSpec a complete `TypeName` since that’d necessarily have to match the `TypeName` it’s being added to.

**Comment by xiemeilong** [comment]
I also need this feature.

**Comment by dharezlak** [comment]
HI, what is the best way to generate a class implementing a generated interface?
